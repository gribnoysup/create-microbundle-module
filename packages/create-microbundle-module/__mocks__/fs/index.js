const path = require('path');
// INFO: We will replace actual file system with
// in-memory implementation
const { vol: volume, fs: memfs } = require('memfs');

const fs = require.requireActual('fs');

// INFO: For this to work properly, we will copy
// all the templates to in-memory volume
const templatesDir = (...pathSegments) =>
  path.resolve(__dirname, '..', '..', 'templates', ...pathSegments);

const templateFiles = fs.readdirSync(templatesDir()).reduce((acc, fileName) => {
  acc[fileName] = fs.readFileSync(templatesDir(fileName), 'utf-8');
  return acc;
}, {});

volume.fromJSON(templateFiles, templatesDir());

module.exports = memfs;
