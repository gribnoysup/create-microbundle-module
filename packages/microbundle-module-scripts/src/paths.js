import path from 'path';
import fs from 'fs';

// Make sure any symlinks in the project folder are resolved:
// https://github.com/facebookincubator/create-react-app/issues/637
// https://github.com/jaredpalmer/razzle/blob/e77c8e7f6410c9a645daedd90c987ad61f7a2392/packages/razzle/config/paths.js#L10
const appDirectory = fs.realpathSync(process.cwd());

const resolveApp = (relativePath = '.', ...args) =>
  path.resolve(appDirectory, relativePath, ...args);

const resolveOwn = (relativePath = '.', ...args) =>
  path.resolve(__dirname, '..', relativePath, ...args);

const resolveSrc = (relativePath = '.', ...args) =>
  path.resolve(__dirname, relativePath, ...args);

const nodePaths = (process.env.NODE_PATH || '')
  .split(process.platform === 'win32' ? ';' : ':')
  .filter(Boolean)
  .filter(folder => !path.isAbsolute(folder))
  .map(resolveApp);

export default {
  nodePaths: nodePaths,
  ownPath: resolveOwn(),
  appPath: resolveApp(),
  appPackageJSONPath: resolveApp('package.json'),
  runnerPath: resolveOwn('index.js'),
  // --- CONFIG:
  ownPrettierConfig: resolveSrc('config', 'prettier.js'),
  appPrettierConfig: resolveApp('prettier.config.js'),
  ownLintStagedConfig: resolveSrc('config', 'lint-staged.js'),
  appLintStagedConfig: resolveApp('lint-staged.config.js'),
  ownEslintConfig: resolveSrc('config', 'eslint.js'),
  appEslintConfig: resolveApp('.eslintrc.js'),
  ownJestConfig: resolveSrc('config', 'jest.js'),
  appJestConfig: resolveApp('jest.config.js'),
  // --- BIN:
  lintStagedBin: require.resolve('lint-staged'),
  jestBin: require.resolve('jest'),
  // INFO: These escape hatches should be used only for testing purposes
  __updateAppPaths(newPath) {
    this.appPath = path.resolve(newPath);
    this.appPrettierConfig = path.resolve(newPath, 'prettier.config.js');
    this.appLintStagedConfig = path.resolve(newPath, 'lint-staged.config.js');
    this.appEslintConfig = path.resolve(newPath, '.eslintrc.js');
  },
  __resetAppPaths() {
    this.__updateAppPaths(appDirectory);
  },
};
