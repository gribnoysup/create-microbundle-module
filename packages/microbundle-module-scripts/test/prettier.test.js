import fs from 'fs-extra';

import paths from '../src/paths';
import prettify, { getPrettierConfig } from '../src/prettier';

import fixturePaths from './__fixtures__/paths';

describe('getPrettierConfig', () => {
  afterEach(() => {
    paths.__resetAppPaths();
  });

  it("should resolve own config when application doesn't have any", async () => {
    paths.__updateAppPaths(fixturePaths.basicModule);
    expect(await getPrettierConfig()).toMatchSnapshot();
  });

  it('should resolve app config when application has one', async () => {
    paths.__updateAppPaths(fixturePaths.withPrettierConfig);
    expect(await getPrettierConfig()).toMatchSnapshot();
  });

  it('should return undefined while trying to prettify non-existent file', async () => {
    const output = await prettify(['/__NO_WAY_THIS_EXISTS__.js']);
    expect(output[0]).toBeUndefined();
  });

  it('should return "prettified" source when file exists', async () => {
    paths.__updateAppPaths(fixturePaths.basicModule);

    const output = await prettify([
      fixturePaths.resolve(fixturePaths.basicModule, 'src', 'index.js'),
    ]);

    expect(output[0]).toMatchSnapshot();
  });

  it('should write file if `write` option is passed', async () => {
    paths.__updateAppPaths(fixturePaths.basicModule);

    const spy = jest.spyOn(fs, 'writeFile');

    const src = fixturePaths.resolve(
      fixturePaths.basicModule,
      'src',
      'index.js'
    );

    const output = await prettify([src], { write: true });

    expect(spy).toBeCalledWith(src, output[0]);

    spy.mockRestore();
  });
});
