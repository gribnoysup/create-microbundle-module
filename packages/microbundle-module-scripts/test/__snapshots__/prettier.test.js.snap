// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPrettierConfig should resolve app config when application has one 1`] = `
Object {
  "customPrettierConfig": true,
}
`;

exports[`getPrettierConfig should resolve own config when application doesn't have any 1`] = `
Object {
  "trailingComma": "es5",
}
`;

exports[`getPrettierConfig should return "prettified" source when file exists 1`] = `
"/* eslint-disable */

export default async function myModule() {
  const result = await Promise.resolve(123);
  return result.toString();
}
"
`;
